src/main/java/org/terasology/changingBlocks/conditional/ConditionalBlocksSystem.java

```diff
@@ -0,0 +1,373 @@
+import org.terasology.changingBlocks.conditional.components.ConditionalBlockChangeComponent;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeRemoveComponent;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnAddedComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.CharacterComponent;
+import org.terasology.logic.inventory.ItemComponent;
+import org.terasology.logic.location.LocationChangedEvent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.math.Direction;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.physics.Physics;
+import org.terasology.physics.StandardCollisionGroup;
+import org.terasology.registry.In;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+import org.terasology.world.OnChangedBlock;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.BlockComponent;
+import org.terasology.world.block.BlockManager;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class ConditionalBlocksSystem extends BaseComponentSystem {
+
+    @In
+    private BlockManager blockManager;
+    @In
+    private EntityManager entityManager;
+    @In
+    private WorldProvider worldprovider;
+    @In
+    private Physics physics;
+    @In
+    private Random random;
+
+    private static Logger log = LoggerFactory.getLogger(ConditionalBlocksSystem.class);
+
+    /**
+     * Maps block IDs and entity categories to lists of block entities that have changes triggered by such blocks or entities.
+     */
+    private Map<String, List<EntityRef>> triggerCollections = new HashMap<>();
+
+    @Override
+    public void initialise() {
+        random = new FastRandom(worldprovider.getSeed().hashCode());
+    }
+
+    /**
+     * Registers a particular block entity with a particular triggering string.
+     * This method is public so that other modules may register their own custom entity-trigger types.
+     * @param trigger The name of the trigger, such as a block ID or entity category.
+     * @param triggerable The entity that may be changed by the trigger.
+     * @param isBlock Whether this trigger is a block or a free-moving entity like a player or NPC.
+     */
+    public void registerTrigger(String trigger, EntityRef triggerable, Boolean isBlock)
```

Comment:
Should we ensure that the trigger string is case insensitive here? At least the internal implementation should work regardless of the case. Could use `Name` class for that.